src/Pages/Repositories/ContentListTable/components/UploadContent/compontents/FileUploader.tsx

```diff
@@ -0,0 +1,331 @@
+          currentFiles[name].failed = true;
+          currentFiles[name].completed = false;
+          setCurrentFiles((prev) => ({ ...prev, [name]: currentFiles[name] }));
+          targetIndexes.clear();
+        }
+      }
+    }
+  };
+
+  const retryItem = (name: string) => {
+    if (!currentFiles[name]) return;
+    currentFiles[name].error = '';
+    currentFiles[name].failed = false;
+    currentFiles[name].chunks = currentFiles[name].chunks.map((chunk) => ({
+      ...chunk,
+      retryCount: 0,
+      // If Chunk is not completed, we know that it failed, so we want to reset it to be requeued
+      queued: chunk.completed,
+    }));
+    setCurrentFiles((prev) => ({ ...prev, [name]: currentFiles[name] }));
+  };
+
+  useEffect(() => {
+    if (!isBatching && !isDropping) {
+      setIsBatching(true);
+      const allDownloads = Object.keys(currentFiles).filter(
+        (name) =>
+          !currentFiles[name].completed &&
+          !currentFiles[name].failed &&
+          currentFiles[name].chunks.some((chunk) => !chunk.queued),
+      );
+      const underBatchSize = allDownloads.filter(
+        (name) => currentFiles[name].file.size < MAX_CHUNK_SIZE,
+      );
+      const overBatchSize = allDownloads.filter(
+        (name) => currentFiles[name].file.size >= MAX_CHUNK_SIZE,
+      );
+      (async () => {
+        while (underBatchSize.length) {
+          const batch = underBatchSize.splice(0, BATCH_SIZE);
+          await Promise.all(batch.map((name) => updateItem(name)));
+        }
+
+        for (let index = 0; index < overBatchSize.length; index++) {
+          const name = overBatchSize[index];
+          await updateItem(name);
+        }
+        setIsBatching(false);
+      })();
+    }
+  }, [fileCount, completedCount, failedCount, isBatching, isDropping]);
+
+  const storeFileInfoForUpdate = async (file: File) => {
+    const totalCount =
+      file.size % MAX_CHUNK_SIZE == 0
+        ? file.size / MAX_CHUNK_SIZE
+        : Math.floor(file.size / MAX_CHUNK_SIZE) + 1;
+
+    const chunks: Chunk[] = [];
+
+    for (let index = 0; index < totalCount; index++) {
+      const start = index ? index * MAX_CHUNK_SIZE : 0;
+      let end = (index + 1) * MAX_CHUNK_SIZE - 1;
+      if (index === totalCount - 1) {
+        end = file.size - 1;
+      }
+
+      chunks.push({ start, end, queued: false, completed: false, retryCount: 0 });
+    }
+
+    let checksum: string;
+    let error: string | undefined = undefined;
+
+    try {
+      checksum = await getFileChecksumSHA256(file);
+    } catch (err) {
+      error = 'Failed checksum validation: ' + (err as Error).message;
+    }
+
+    let uuid: string = '';
+    let created: string = '';
+    if (!error) {
+      try {
+        const res = await createUpload(file.size);
+        uuid = res.upload_uuid;
+        created = res.created;
+      } catch (err) {
+        error = 'Failed checksum validation: ' + (err as Error).message;
```

Comment:
during a network outage i sometimes see this message: `Failed checksum validation: Network Error`. i don't think we need `Failed checksum validation` in the error message in this case

Reply:
Good catch, the retry logic needed some work!